pdns/dnsdist-lua-actions.cc

```diff
@@ -540,21 +540,28 @@ DNSAction::Action SpoofAction::operator()(DNSQuestion* dq, std::string* ruleresu
   uint16_t qtype = dq->qtype;
   // do we even have a response?
   if (d_cname.empty() &&
-      d_rawResponse.empty() &&
+      d_rawResponses.empty() &&
       d_types.count(qtype) == 0) {
     return Action::None;
   }
 
   vector<ComboAddress> addrs;
+  vector<std::string> rawResponses;
   unsigned int totrdatalen = 0;
   uint16_t numberOfRecords = 0;
   if (!d_cname.empty()) {
     qtype = QType::CNAME;
     totrdatalen += d_cname.getStorage().size();
     numberOfRecords = 1;
-  } else if (!d_rawResponse.empty()) {
-    totrdatalen += d_rawResponse.size();
-    numberOfRecords = 1;
+  } else if (!d_rawResponses.empty()) {
+    for(const auto& rawResponse : d_rawResponses){
```

Comment:
We could call `rawResponses.reserve(d_rawResponses.size());` to make sure we do only one allocation, no matter the number of items.

Reply:
added